Frontend/ui-driver/src/Components/BottomNavBar/Controller.purs

```diff
@@ -16,63 +16,69 @@
-    }
-    ]) <> 
-    [{
-      activeIcon: "ny_ic_alerts_active",
-      defaultIcon: "ny_ic_alerts_inactive," <> (getCommonAssetStoreLink FunctionCall) <> "ny_ic_alerts_inactive.png",
-      text: "Alert"
-    }
-  ]
-}
+navData :: ScreenNames.ScreenName -> BottomNavConfig -> BottomNavBarState
+navData screenName bottomNavConfig = do
+  let showNewBannerOnSubscription = (Maybe.fromMaybe 0 $ fromString $ getValueToLocalNativeStore TIMES_OPENED_NEW_SUBSCRIPTION) < 3
+      navdata = [
+        {
+          activeIcon: "ny_ic_home_active," <> (getCommonAssetStoreLink FunctionCall) <> "ny_ic_home_active.png",
+          defaultIcon: "ny_ic_home_inactive," <> (getCommonAssetStoreLink FunctionCall) <> "ny_ic_home_inactive.png",
+          text: "Home",
+          showNewBanner : bottomNavConfig.home.showNew,
+          isVisible : bottomNavConfig.home.isVisible,
+          screenName : ScreenNames.HOME_SCREEN
+        },
+        {
+          activeIcon: "ny_ic_rides_tab_active," <> (getAssetStoreLink FunctionCall) <> "ny_ic_rides_tab_active.png",
+          defaultIcon: "ny_ic_rides_tab_inactive," <> (getAssetStoreLink FunctionCall) <> "ny_ic_rides_tab_inactive.png",
+          isVisible : bottomNavConfig.rideHistory.isVisible,
+          showNewBanner : bottomNavConfig.rideHistory.showNew,
+          text: "Rides",
+          screenName : ScreenNames.RIDE_HISTORY_SCREEN
+        },
+        {
+          activeIcon: "ny_ic_join_active," <> (getAssetStoreLink FunctionCall) <> "ny_ic_join_active.png",
+          defaultIcon: "ny_ic_join_inactive," <> (getAssetStoreLink FunctionCall) <> "ny_ic_join_inactive.png",
+          isVisible : bottomNavConfig.subscription.isVisible,
+          showNewBanner : bottomNavConfig.subscription.showNew && showNewBannerOnSubscription ,
+          text: "Join",
+          screenName : ScreenNames.SUBSCRIPTION_SCREEN
+        },
+        {
+          activeIcon: "ic_referral_active," <> (getAssetStoreLink FunctionCall) <> "ic_referral_active.png",
+          defaultIcon: if (getValueToLocalNativeStore REFERRAL_ACTIVATED) == "true" then  "ny_ic_contest_alert," <> (getCommonAssetStoreLink FunctionCall) <> "ny_ic_contest_alert.png" else "ic_referral_inactive," <> (getCommonAssetStoreLink FunctionCall) <> "ic_referral_inactive.png",
+          isVisible : bottomNavConfig.referral.isVisible,
+          showNewBanner : bottomNavConfig.referral.showNew,
+          text: "Rankings",
+          screenName : ScreenNames.REFERRAL_SCREEN
+        },
+        {
+          activeIcon: "ny_ic_alerts_active",
+          defaultIcon: "ny_ic_alerts_inactive," <> (getCommonAssetStoreLink FunctionCall) <> "ny_ic_alerts_inactive.png",
+          text: "Alert",
+          isVisible : bottomNavConfig.notifications.isVisible,
+          showNewBanner : bottomNavConfig.notifications.showNew,
+          screenName : ScreenNames.ALERTS_SCREEN
+        }
+      ]
+      processedNavOptions = DA.filter (\item -> item.isVisible == true) navdata
```

Comment:
No need to compare boolean
Can be written as DA.filter (\item -> item.isVisible) navdata